python-client/giskard/core/validation.py

```diff
@@ -1,4 +1,23 @@
 import pandas as pd
+from pydantic import validate_arguments
+import functools
+
+
+def validate_args(func):
```

Comment:
since you're not modifying the logic of the underlying original function, you can skip the inner method creation (`def foncigured_decorator`) and just create your `validate_args` like
```
def validate_args(func):
    """
    Decorator to enforce a function args to be compatible with their type hints.
    :return: A wrapper function decorated by pydantic validate_arguments configured to allow arbitrary types check.
    """
    return functools.wraps(func)(
        validate_arguments(config=dict(arbitrary_types_allowed=True))(func)
    )
```

nitpick: I'd also name it after the original decorator - configured_decorator so that they're interchangeable (only the import will have to be changed in case you want to use one or another)

Reply:
done!